tests/pipeline/nodes/model/fairmotv1/test_fairmot.py

```diff
@@ -0,0 +1,314 @@
+        self, test_human_video_sequences, fairmot_config
+    ):
+        _, detections = test_human_video_sequences
+        fairmot = Node(fairmot_config)
+        # Make frame_id start at 2 internally to avoid STrack from activating
+        # in activate() when frame_id == 1
+        fairmot.model.tracker.frame_id = 1
+        prev_tags = []
+        for i, inputs in enumerate({"img": x["img"]} for x in detections):
+            output = fairmot.run(inputs)
+            if i > 1:
+                assert output["obj_tags"] == prev_tags
+            prev_tags = output["obj_tags"]
+
+    def test_reactivate_tracks(self, test_human_video_sequences, fairmot_config):
+        _, detections = test_human_video_sequences
+        fairmot = Node(fairmot_config)
+        prev_tags = []
+        for i, inputs in enumerate({"img": x["img"]} for x in detections):
+            if i == SEQ_IDX:
+                # These STrack should get re-activated
+                for track in fairmot.model.tracker.tracked_stracks:
+                    track.mark_lost()
+            output = fairmot.run(inputs)
+            if i > 1:
+                assert output["obj_tags"] == prev_tags
+            prev_tags = output["obj_tags"]
+
+    def test_associate_with_iou(self, test_human_video_sequences, fairmot_config):
+        _, detections = test_human_video_sequences
+        fairmot = Node(fairmot_config)
+        prev_tags = []
+        with mock.patch(
+            "peekingduck.pipeline.nodes.model.fairmotv1.fairmot_files.matching.fuse_motion",
+            wraps=replace_fuse_motion,
+        ):
+            for i, inputs in enumerate({"img": x["img"]} for x in detections):
+                output = fairmot.run(inputs)
+                if i > 1:
+                    assert output["obj_tags"] == prev_tags
+                prev_tags = output["obj_tags"]
+
+    def test_mark_unconfirmed_tracks_for_removal(
+        self, test_human_video_sequences, fairmot_config
+    ):
+        """Manipulate both embedding and iou distance to be above the cost
+        limit so nothing gets associated and all gets marked for removal. As a
+        result, the Tracker should no produce any track IDs.
+        """
+        _, detections = test_human_video_sequences
+        fairmot = Node(fairmot_config)
+        with mock.patch(
+            "peekingduck.pipeline.nodes.model.fairmotv1.fairmot_files.matching.fuse_motion",
+            wraps=replace_fuse_motion,
+        ), mock.patch(
+            "peekingduck.pipeline.nodes.model.fairmotv1.fairmot_files.matching.iou_distance",
+            wraps=replace_iou_distance,
+        ):
+            for i, inputs in enumerate({"img": x["img"]} for x in detections):
+                output = fairmot.run(inputs)
+                if i == 0:
+                    # Skipping the assert on the first frame FairMOT sets
+                    # STrack to is_activated=True on when frame_id=1 but JDE
+                    # doesn't
```

Comment:
Maybe add a punctuation between "frame FairMOT" so that comment reads better

Reply:
Added